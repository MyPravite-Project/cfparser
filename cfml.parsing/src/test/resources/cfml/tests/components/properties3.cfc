component accessors="true"{
  
  property obj.infrastructure.content.NewsRepository newsRepository;

}