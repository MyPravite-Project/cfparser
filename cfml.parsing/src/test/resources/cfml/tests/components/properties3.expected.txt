/*===TOKENS===*/
COMPONENT                <component>
IDENTIFIER               <accessors>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <true>
CLOSE_STRING             <">
'{'                      <{>
PROPERTY                 <property>
IDENTIFIER               <obj>
'.'                      <.>
IDENTIFIER               <infrastructure>
'.'                      <.>
CONTENT                  <content>
'.'                      <.>
IDENTIFIER               <NewsRepository>
IDENTIFIER               <newsRepository>
';'                      <;>
'}'                      <}>
/*===TREE===*/
(scriptBlock
  (componentDeclaration
    component
    (componentAttribute
      (identifier accessors)
      =
      (startExpression
        (compareExpression
          (baseExpression
            (unaryExpression
              (primaryExpression
                (literalExpression (stringLiteral " (stringLiteralPart true) "))
              )
            )
          )
        )
      )
    )
    (componentGuts
      {
      (element
        (statement
          (tagOperatorStatement
            (propertyStatement
              property
              (typeSpec
                (multipartIdentifier
                  (identifier obj)
                  .
                  (identifierOrReservedWord (identifier infrastructure))
                  .
                  (identifierOrReservedWord (identifier (cfmlFunction content)))
                  .
                  (identifierOrReservedWord (identifier NewsRepository))
                )
              )
              (identifier newsRepository)
              ;
            )
          )
        )
      )
      }
    )
  )
)
/*======*/
/*===DECOMPILE===*/
component accessors='true' {
property obj.infrastructure.content.NewsRepository newsRepository
}
/*======*/